apps/web/src/components/transactions/TxDetails/TxData/DecodedData/SingleTxDecoded/index.tsx

```diff
@@ -30,6 +40,54 @@ export const SingleTxDecoded = ({ tx, txData, actionTitle, variant, expanded, on
 
   const safeToL2MigrationDeployment = getSafeToL2MigrationDeployment()
   const safeToL2MigrationAddress = chain && safeToL2MigrationDeployment?.networkAddresses[chain.chainId]
+  const tokenInfoIndex = (txData as TransactionDetails['txData'])?.tokenInfoIndex
+
+  const txDataHex = tx.data ?? txData.hexData
+
+  const isTransfer = Boolean(txDataHex?.startsWith(ERC20_INTERFACE.getFunction('transfer').selector))
+
+  const transferInfos:
+    | {
+        recipient: string
+        transferValue: string
+        tokenInfo: Erc20Token | NativeToken
+      }
+    | undefined = useMemo(() => {
+    if ((!isTransfer && !isNativeTransfer) || !txDataHex || !chain) {
+      return
+    }
+    try {
+      if (isTransfer) {
+        const [recipient, transferValue] = ERC20_INTERFACE.decodeFunctionData('transfer', txDataHex)
+        const tokenInfo = isTransfer ? tokenInfoIndex?.[tx.to] : undefined
+
+        if (tokenInfo?.type !== 'ERC20') {
+          return
+        }
+
+        return { recipient, transferValue, tokenInfo }
+      }
+
+      if (!tx.value || tx.value === '0') {
+        return
+      }
+
+      return {
+        recipient: tx.to,
+        transferValue: tx.value,
+        tokenInfo: {
+          type: 'NATIVE_TOKEN',
+          address: ZERO_ADDRESS,
+          decimals: chain.nativeCurrency.decimals,
+          symbol: chain.nativeCurrency.symbol,
+          logoUri: chain.nativeCurrency.logoUri,
+          name: chain.nativeCurrency.name,
+        },
```

Comment:
This should [already be being returned](https://github.com/safe-global/safe-client-gateway/pull/2537/files#diff-4604e6df22d648e6ce0075b253a5eeb2007ec715c3dd5e512c9653a6de861231R234-R246) and not need to be mapped. Did you encounter issues?

Reply:
Formatting the native tokens in the frontend is usually done with this `tokenInfo`.
I see the advantage that it cannot be `undefined`.
I could make it reusable through a `useNativeTokenInfo` hook though and use it across the app.